track/track_test.go

```diff
@@ -0,0 +1,76 @@
+package track
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrack(t *testing.T) {
+
+	t.Run("Track 1 data", func(t *testing.T) {
+		tracker := NewTrackFirst(true)
+
+		samples := []string{
+			`B4815881002867896^YATES/EUGENE JOHN         ^21129821000123456789`,
+			`B4815881002861896^YATES/EUGENE L            ^^^356858      00998000000`,
+			`B4000340000000506^John/Doe                  ^13011110000123000`,
+			`B1234567890123445^PADILLA/L.                ^99011200000000000000**XXX******`,
+		}
+		for _, sample := range samples {
+			card, err := tracker.Read([]byte(sample))
+			require.NoError(t, err)
+			buf, err := tracker.Write(card)
+			require.NoError(t, err)
+			require.Equal(t, sample, string(buf))
+		}
+	})
+	t.Run("Track 1 data with invalid name length", func(t *testing.T) {
+		tracker := NewTrackFirst(false)
+
+		samples := []string{
+			`B4242424242424242^SMITH JOHN Q^11052011000000000000`,
+		}
+		for _, sample := range samples {
+			card, err := tracker.Read([]byte(sample))
+			require.NoError(t, err)
+			buf, err := tracker.Write(card)
+			require.NoError(t, err)
+			require.Equal(t, sample, string(buf))
+		}
+	})
+
+	t.Run("Track 2 data", func(t *testing.T) {
+		tracker := NewTrackSecond()
+
+		samples := []string{
+			`4000340000000506=2512111123400001230`,
+			`4242424242424242=15052011000000000000`,
+			`1234567890123445=99011200XXXX00000000`,
+		}
+		for _, sample := range samples {
+			card, err := tracker.Read([]byte(sample))
```

Comment:
As in my previous comment, we need to check if we parse values properly. As with table tests I would use `input` and `want` structure with expected values.